04/app.js

```diff
@@ -6,68 +6,59 @@ const lang ='pl';
-    const latValue = document.querySelector('.weather__lat');
-    const lngValue = document.querySelector('.weather__lng');
-    latValue.innerText = lat;
-    lngValue.innerText = lng
+    const latValue = document.querySelector('.weather__lat').innerText=lat;
+    const lngValue = document.querySelector('.weather__lng').innerText=lng;
 }
 
 const addWeatherInfo = (response) => {
-    //const {data:[{city_name},{weather:{description}}]} = response; // problem z tym zapisem, a wydaje mi sie ze jest ok (odpowiednie nawiasy i kolejnosc),chciałam w jednym miejscu pobrac wszystkie dane, to w ogole mozliwe? Strasznie mnie nurtuje ten problem , czytanie  bledzie mi nie pomoglo - blad Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'weather')
-    const {data:[{city_name}]} = response;
-    const {data:[{weather:{description}}]} = response;
-    const {data:[{temp}]} = response;
+    const {data:[{city_name,weather:{description},temp}]} = response; 
     const fahrenheit = Math.round((temp * 1.8) + 32);
-
     const spanCity = document.querySelector('.city__name').innerText = city_name;
     const spanDesc = document.querySelector('.weather__summary').innerText = description;
     const spanTemp = document.querySelector('.weather__temperature').innerText = fahrenheit;
 }
 
-function getDataFromUser(){
-    const formEl  = document.querySelector('form');
-    formEl.addEventListener('submit', e => {
-        e.preventDefault();
-        const inputLat = e.target.querySelector('.form__field--lat').value;
-        const inputLng = e.target.querySelector('.form__field--lng').value;;
-        const coordinatesData = {lat:inputLat,lng:inputLng}
-        addCoordinatesText(coordinatesData);
-        takeWeatherData(coordinatesData);
-    });
+function getDataFromUser(e){
+    e.preventDefault();
+    const inputLat = e.target.querySelector('.form__field--lat').value;
+    const inputLng = e.target.querySelector('.form__field--lng').value;;
+    const coordinatesData = {lat:inputLat,lng:inputLng}
+    addCoordinatesText(coordinatesData)
+    takeWeatherData(coordinatesData);
 }
 
 function takeWeatherData({lat,lng}) {
-   fetch(`${apiUrl}?key=${apiKey}&lang=${lang}&lat=${lat}&lon=${lng}`)
-        .then(response => {
-            if(response.ok) {return response.json();}
-            return new Promise.reject(response)
-        })
-        // .then(response => addCityName(response))
-        // .then(response => addWeatherDescription(response))
-        .then(response => addWeatherInfo(response))
-        .catch(error => console.error(error))
-}
+    fetch(`${apiUrl}?key=${apiKey}&lang=${lang}&lat=${lat}&lon=${lng}`)
+         .then(response => {
+             if(response.ok) {return response.json();}
+             return new Promise.reject(response)
+         })
+         .then(response => addWeatherInfo(response))
+        //  .then(response=> addCityName(response))
+        //  .then(response => console.log(response))
```

Comment:
Zwróć uwagę, że ten `.then()` "łamie łańcuch", musi być tylko tyle, aby to zadziałało :)
```
    fetch(`${apiUrl}?key=${apiKey}&lang=${lang}&lat=${lat}&lon=${lng}`)
         .then(response => {
             if(response.ok) {return response.json();}
             return new Promise.reject(response)
         })
         .then(response=> addCityName(response))
         .then(response => addWeatherDescription(response))
         .catch(error => console.error(error)
```